xds/server.go

```diff
@@ -317,18 +322,66 @@ func (s *GRPCServer) GracefulStop() {
 	}
 }
 
+func routeAndProcess(ctx context.Context) error {
```

Comment:
Could you please add a comment describing what this function does.

Reply:
Also, this function is the meat of this PR and I don't see any tests for this.

Reply:
Ah, sorry, I see the tests now.

Reply:
Added comment.